src/dune_rules/toplevel.mli

```diff
@@ -15,7 +15,9 @@ end
 
 type t
 
-val setup_rules : t -> unit Memo.Build.t
+(** If [is_implicit] is [true], do not produce a symlink to the parent
+    directory. *)
+val setup_rules : t -> is_implicit:bool -> unit Memo.Build.t
```

Comment:
What about a more concrete argument name like `symlink_parent`?

Reply:
Sounds good. In fact, this argument is not needed.  This function is used twice: once with `false` and once with `true` internally. I moved the setting up of the symlink in the one place it was set to `true`. Seems more modular this way.

BTW, do you remember why you exposed `Toplevel.t`? The only place it is used is to do `make` immediately followed by `setup_rules`. Seems to me that we could expose only `setup_rules`.

Reply:
>  Seems to me that we could expose only setup_rules.

That should work.